src/main/java/com/project/gamestore/services/UserGameService.java

```diff
@@ -57,7 +57,7 @@ public UserGameDTO update(UserGameDTO userGameDTO, UUID publicIdentifier) {
 
         Game game = gameRepository.findByPublicIdentifier(userGameDTO.getGameId()).orElseThrow();
```

Comment:
You also need to change this to use `findByPublicIdentifierMandatory`. Please check everywhere you use `findByPublicIdentifier` and modify accordingly